### PR TITLE
Refactored everything

### PR DESCRIPTION
**Pull request by MarcelCoding**
This pull request implements features from #12

### CODE REVIEW

File: src/_core/map/map.component.ts [hunk 1]
```diff
@@ -28,15 +40,25 @@ export interface Location {
...
-  private map: OlMap | undefined;
+  private map?: Map;
+  private vectorSource?: VectorSource;
+
+  private readonly vehicles: Record<string, Vehicle<Geometry>> = {};
 
-  private readonly location0: Subject<Location> = new Subject<Location>();
-  public readonly location = this.location0.pipe(distinct(), debounceTime(50));
-  private readonly features: Map<string, Feature<Geometry>> = new Map<string, Feature<Geometry>>();
+  private readonly location0 = new Subject<Location>();
```
Comment (revol-xut) [inline src/_core/map/map.component.ts:48]: why is there location0 and location ?

File: src/_core/map/map.component.ts [hunk 2]
```diff
@@ -58,60 +80,59 @@ export class MapComponent implements OnInit, OnDestroy {
...
+    this.vectorSource?.addFeature(vehicle.feature);
   }
 
-  public removeFeature(id: string): void {
-    const existing = this.features.get(id);
-    if (existing) {
-      this.features.delete(id);
-      this.vectorSource?.removeFeature(existing);
-    }
+  public getVehicle(id: string): Vehicle<Geometry> | undefined {
```
Comment (revol-xut) [inline src/_core/map/map.component.ts:106]: why this undefined ?

File: src/_core/ws.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import {Observable, Subject} from "rxjs";
+
+export class WsSocket<I, O> {
+
+  private readonly stream0 = new Subject<O>();
```
Comment (revol-xut) [inline src/_core/ws.ts:5]: stream0 ?

File: src/app/pages/region/region-map/region-map.component.ts [hunk 4]
```diff
@@ -39,132 +36,164 @@ export class RegionMapComponent implements AfterViewInit, OnDestroy {
...
+      ).subscribe(data => {
+        const id = `${data.line}_${data.run}`;
+
+        const vehicle = this.map.getVehicle(id);
+        if (vehicle) {
+          console.log("update");
+          vehicle.move(data.coords, data.delay);
+        }
+        else {
+          console.log("new");
```
Comment (revol-xut) [inline src/app/pages/region/region-map/region-map.component.ts:70]: more verbose print perhaps

File: src/app/pages/region/region-map/region-map.component.ts [hunk 5]
```diff
@@ -39,132 +36,164 @@ export class RegionMapComponent implements AfterViewInit, OnDestroy {
...
+              case LineType.BUS:
+                this.map.addVehicle(new BusVehicle(id, line?.name ?? `(${data.line})`, line?.description, data.coords));
+                break;
+            }
+          }
+        }
       });
   }
 
+  // concat(
```
Comment (revol-xut) [inline src/app/pages/region/region-map/region-map.component.ts:90]: what happens with this dead code ?

**Review by revol-xut — CHANGES_REQUESTED** [review]

**Comment by revol-xut** [comment]
More of questions for my understanding :D 

Thank you very much

File: src/app/pages/region/region-map/region-map.component.ts (hunk 4 above)
Comment (MarcelCoding) [reply src/app/pages/region/region-map/region-map.component.ts:70]: thats remaining debugging and should be removed, by bad

File: src/app/pages/region/region-map/region-map.component.ts (hunk 5 above)
Comment (MarcelCoding) [reply src/app/pages/region/region-map/region-map.component.ts:90]: This got replayed by the new implmentation or is remaining stuff from the interpolation, which is still missing in the new one.

File: src/_core/map/map.component.ts (hunk 1 above)
Comment (MarcelCoding) [reply src/_core/map/map.component.ts:48]: location0 is the action (private) field that holds the data and location is a public view to the data. (different "access permissions")

File: src/_core/map/map.component.ts (hunk 2 above)
Comment (MarcelCoding) [reply src/_core/map/map.component.ts:106]: if the id could not be found in the provided data it is unknown and will be displayed as a car and not a tram or bus.

File: src/_core/ws.ts (hunk 3 above)
Comment (MarcelCoding) [reply src/_core/ws.ts:5]: the is a public field `stream` that is a view of the prvate `stream0` field

**Review by revol-xut — APPROVED** [review]

File: src/_core/map/map.component.ts (hunk 1 above)
Comment (revol-xut) [reply src/_core/map/map.component.ts:48]: maybe rename it public_location and private_location for verbosity

**Comment by revol-xut** [comment]
@MarcelCoding Do want to merge this ?

**Comment by MarcelCoding** [comment]
Yes but icons

**Comment by MarcelCoding** [comment]
I need to add all possible shades and improve the tram

**Pull request merged**